docs/v1/P4Runtime-Spec.adoc

```diff
@@ -3897,6 +3901,38 @@ is `sum_of_members`, or else the server must return `INVALID_ARGUMENT`. Each
   on the `watch_port` field, which also apply for one shot action selector
   programming.
 
+Each `ActionSelectionMode` enum can take on the following values:
+
+* `DEFAULT_MODE_DETERMINED_BY_ACTION_SELECTOR`: uses the action specified by the
+  action selector, ensuring backward compatibility. In a SAI-based switch, this
+  would correspond to setting `SAI_NEXT_HOP_GROUP_ATTR_HASH_ALGORITHM` to
+  `SAI_HASH_ALGORITHM_NONE`, using the default action selector mode.
+
+* `HASH`: selects an action based on the result of the hashing algorithm chosen
+  by the switch.
+
+* `RANDOM`: selects an action at random. In a SAI-based switch, this would
+  correspond to setting `SAI_NEXT_HOP_GROUP_ATTR_HASH_ALGORITHM` to
+  `SAI_HASH_ALGORITHM_RANDOM`.
+
+Each `SizeSemantics` enum can take on the following values:
+
+* `DEFAULT_SIZE_DETERMINED_BY_ACTION_SELECTOR`: uses the size semantics
+  specified by the action selector, ensuring backward compatibility.
+
+* `SUM_OF_WEIGHTS`: the group's size is the sum of the individual weights of
+  all its members.
+
+* `SUM_OF_MEMBERS`: the group's size is the total count of its members.
```

Comment:
@jafingerhut, do you think it would be useful to have a SAI correspondence here too? The SUM_OF_MEMBERS one hasn't been upstreamed yet (though they're apparently working on it), but there are a few that correspond to `SUM_OF_WEIGHTS` I believe.

Reply:
Any correspondences / links to SAI documentation of these features seems useful to me.